### ISSUE TITLE
Implement feature to allow upgrade from resin2.x to resin2.x version

### ISSUE THREAD

**Issue by shaunmulligan** [resin-os/resinhup#58]
We should be able to upgrade/downgrade between 2.X versions

<img src="https://frontapp.com/assets/img/icons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_1ta9)

**Comment by agherzan** [comment]
This is blocked by the 2.0 release. Otherwise we will need to invest time in repartitioning due to partition changes in the beta versions. @petrosagg Do we consider including support for upgrading the beta versions too?

**Comment by shaunmulligan** [comment]
@agherzan I don't think we need to worry about upgrading beta versions as they should not be used in production . We just need to be able to go from 2.0.0-prod to 2.1.0- prod ,

**Comment by agherzan** [comment]
Got it. In this case this task is not needed for production ready.

**Comment by shaunmulligan** [comment]
@agherzan it's not strictly needed, I just see a case where someone sends a device out as soon as we announce its ready and then a week later we release a new version with fixes they really need. Then we have to rush this ticket just for that case

**Comment by agherzan** [comment]
I agree. We should keep it close to the release.

**Comment by shaunmulligan** [comment]
Sweet :+1:

**Comment by pcarranzav** [comment]
I think this connects to https://github.com/resin-io/hq/issues/759

**Comment by willnewton** [comment]
There is a pull request open for this functionality here: https://github.com/resin-os/resinhup/pull/78

**Comment by imrehg** [comment]
Initial version is merged, moving onto testing phase.
